File: src/main/java/itmo/is/project/repository/spaceship/SpaceshipRepository.java
package itmo.is.project.repository.spaceship;

import itmo.is.project.model.spaceship.Spaceship;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface SpaceshipRepository extends JpaRepository<Spaceship, Integer> {
}

File: src/main/kotlin/itmo/isproject/shared/user/repository/SpaceshipRepository.kt
package itmo.isproject.shared.user.repository

import itmo.isproject.shared.user.model.Spaceship
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface SpaceshipRepository : JpaRepository<Spaceship, Int> {

    fun findByPilotId(pilotId: Int?): Spaceship?

    fun findByPilotUsernameInternal(username: String?): Spaceship?
}
